packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -43,5 +44,24 @@ export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
         },
       };
     },
+    getPurpose: async (
+      purposeId: PurposeId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Purpose> => {
+      logger.info(`Retrieving purpose with id ${purposeId}`);
+
+      const { data } = await clients.purposeProcessClient.getPurpose({
```

Comment:
This function can also throw an `organizationNotAllowed` error that will passthrough to the caller.
In order to align the message to other similar errors (see https://github.com/pagopa/interop-be-monorepo/pull/1848#discussion_r2080038551) I would go and refactor that error in `tenantNotAllowed` and rephrase error description / title to say "Tenant" and not "Organization"